src/OrchardCore.Modules/OrchardCore.BackgroundTasks/Controllers/BackgroundTaskController.cs

```diff
@@ -63,22 +62,44 @@ public async Task<IActionResult> Index(AdminIndexOptions options, PagerParameter
 
             var document = await _backgroundTaskManager.GetDocumentAsync();
 
-            var items = _backgroundTasks.Select(task => new BackgroundTaskEntry() { Settings = GetSettings(task, document) });
+            var items = _backgroundTasks.Select(task =>
+            {
+                var defaultSettings = task.GetDefaultSettings();
+
+                if (document.Settings.TryGetValue(task.GetTaskName(), out var settings))
+                {
+                    return new BackgroundTaskEntry()
+                    {
+                        Title = defaultSettings.Title,
+                        Name = settings.Name,
+                        Description = settings.Description,
+                        Enable = settings.Enable,
+                    };
+                }
+
+                return new BackgroundTaskEntry()
+                {
+                    Title = defaultSettings.Title,
+                    Name = defaultSettings.Name,
+                    Description = defaultSettings.Description,
+                    Enable = defaultSettings.Enable,
+                };
+            });
```

Comment:
```csharp
var items = _backgroundTasks.Select(task =>
{
    var defaultSettings = task.GetDefaultSettings();

    var taskEntry = new BackgroundTaskEntry { Title = defaultSettings.Title };

    if (document.Settings.TryGetValue(task.GetTaskName(), out var settings))
    {
        taskEntry.Name = settings.Name;
        taskEntry.Description = settings.Description;
        taskEntry.Enable = settings.Enable;
    }
    else
    {
        taskEntry.Name = defaultSettings.Name;
        taskEntry.Description = defaultSettings.Description;
        taskEntry.Enable = defaultSettings.Enable;
    }
});
```

Reply:
May be you could assign the `settings` directly if the value exists on the document settings

Reply:
I don't understand the ask.

Reply:
@hishamco `taskEntry` is of `BackgroundTaskEntry` type where as `settings` is of `BackgroundTaskSettings`

Reply:
Leave my last comment, coz I saw you did something similar in your PR, anyhow the above snippet might much cleaner